pkg/crc/machine/machine.go

```diff
@@ -212,11 +217,23 @@ func getDriverOptions(machineConfig config.MachineConfig) interface{} {
 	return driver
 }
 
-func setMachineLogging(logging bool) {
-	if !logging {
-		log.SetOutWriter(ioutil.Discard)
-		log.SetErrWriter(ioutil.Discard)
+func setMachineLogging(logs bool) error {
+	if !logs {
+		log.SetDebug(true)
+		logging.CloseLogFile()
+		logfile, err := logging.OpenLogfile()
+		if err != nil {
+			return err
+		}
+		log.SetOutWriter(logfile)
+		log.SetErrWriter(logfile)
 	} else {
 		log.SetDebug(true)
 	}
+	return nil
+}
+
+func unsetMachineLogging() {
+	logging.CloseLogFile()
+	logging.InitLogrus(constants.LogLevel)
```

Comment:
No need to initialize logrus again here..

Reply:
If you don't initialize it again then after you close the logging file it will not have other log info after the start.

Reply:
We initialize logrus in pre run of `root.go` so it should have..

Reply:
> again

I am wondering about this also. This sounds like a problem or else an easy way to mess things up.